lgc/patch/VertexFetch.cpp

```diff
@@ -668,18 +669,23 @@ bool LowerVertexFetch::runImpl(Module &module, PipelineState *pipelineState) {
 // =====================================================================================================================
 // Generate vertex fetch function.
 //
-// @param is64bitFetch : Whether is 64 bit fetch
+// @param bitWidth : bit width of vertex
 // @param [in/out] module : Module
 // @returns : Generated function
-Function *VertexFetchImpl::generateFetchFunction(bool is64bitFetch, Module *module) {
+Function *VertexFetchImpl::generateFetchFunction(unsigned bitWidth, Module *module) {
   BuilderBase builder(*m_context);
   // Helper to create basic block
   auto createBlock = [&](Twine blockName, Function *parent) {
     return BasicBlock::Create(*m_context, blockName, parent);
   };
 
+  bool is16bitFetch = bitWidth == 16;
+  bool is64bitFetch = bitWidth == 64;
+
   auto fetch64Type = FixedVectorType::get(Type::getInt32Ty(*m_context), 2);
-  auto fetchType = FixedVectorType::get(Type::getInt32Ty(*m_context), 4);
+  auto fetch32Type = FixedVectorType::get(Type::getInt32Ty(*m_context), 4);
+  auto fetch16Type = FixedVectorType::get(Type::getHalfTy(*m_context), 4);
```

Comment:
I would expect you to use `i16` here to be consistent. Is there a reason for using `half` instead?

Reply:
No, `i16` should be better. I will update it. Thanks.